### PR TITLE
Mc us23 26

### PR DESCRIPTION
**Pull request by Mariana-21**
## Spec/Features ##
* registered_user_can_checkout_spec

## Routes ## 
* post route for cart increment
* patch route for cart decrement 

## Controllers ## 
* decrement method for cart controller created 
* increment method for cart controller created 
* create method for order controller modify controller created 

## Views ##
* "+" for increment for cart show page
* "-" for decrement for cart show page

### CODE REVIEW

File: app/controllers/cart_controller.rb [hunk 1]
```diff
@@ -22,5 +24,12 @@ def remove_item
     redirect_to '/cart'
   end
 
-
+  def decrement
+    if !cart.quantity_one?(params[:item_id])
```
Comment (perryr16) [inline app/controllers/cart_controller.rb:28]: nice helper

File: spec/features/cart/registered_user_can_checkout_spec.rb [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+    
+    it "And I click the button or link to check out and fill out order info and click create order" do 
+
+      visit "/cart"
+      click_on "Checkout"
+
+      expect(current_path).to eq("/orders/new")
+
+      fill_in :name,	with: "Mike Dao" 
+      fill_in :address,	with: "123 Test Street"
```
Comment (perryr16) [inline spec/features/cart/registered_user_can_checkout_spec.rb:38]: spacing...

File: spec/features/cart/show_spec.rb [hunk 3]
```diff
@@ -55,21 +55,68 @@
...
+        within "##{@paper.id}-quantity" do 
+          expect(page).to have_content("3")
+        end
       end
 
+      it "I can decrement my cart item by one" do
+        visit "/cart"
+
+        within "#cart-item-#{@tire.id}" do
+          within("td:nth-child(4)") do
```
Comment (perryr16) [inline spec/features/cart/show_spec.rb:83]: this is new to me. not sure what td:nth-child does but it looks great

File: app/controllers/cart_controller.rb [hunk 4]
```diff
@@ -22,5 +24,12 @@ def remove_item
...
+      cart.remove_item(params[:item_id])
+      redirect_to "/cart"
+    elsif cart.quantity_one?(params[:item_id])
+      remove_item
+    end
+  end
```
Comment (dborski) [inline app/controllers/cart_controller.rb:34]: Useful helper method

**Pull request merged**